W-19804084-rtf-3.0-documentation-kt

kevintroller opened the pull request:
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

kevintroller committed:
Update antora.yml configuration

kevintroller committed:
Add missing anchor for Create a Connected App section and consolidate volume configuration intro

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+Anypoint Controller is a Kubernetes operator that enables you to deploy and manage Mule applications on Runtime Fabric using GitOps workflows and Custom Resources (CR).
+
+Key Benefits:
```
```suggestion
With Anypoint Controller, you can:
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+* Use standard Kubernetes tools and practices to deploy Mule applications, including `kubectl`, Helm, and custom resources.
+* Leverage Kubernetes features such as external secret managers, volume mounts, and RBAC policies to meet enterprise security requirements.
+
+This guide shows you how to:
```
Avoid the self referential "This guide".

Suggest using this list as a high-level process overview (numbered list with links to the docs where it's covered). See this section for an example: https://docs.mulesoft.com/api-governance/get-started#govern-your-apis 

Check out the overview guidelines, too: https://confluence.internal.salesforce.com/spaces/MTDT/pages/458629222/Writing+Product+Overviews

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+This guide shows you how to:
+
+* Install and configure Anypoint Controller in your Runtime Fabric cluster.
+* Deploy Mule applications using declarative MuleApplication Custom Resources.
```
```suggestion
* Deploy Mule applications using declarative Mule application Custom Resources.
```

hannanelson replied:
(Not sure if that's correct.)

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+[NOTE]
+This feature is currently in Limited General Availability (Limited GA). To enable Anypoint Controller for your organization, contact MuleSoft Support.
+
+Supported features:
```
Are these features? Or are these prerequisites or compatibility?

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+* Runtime Fabric Mule application deployments
+* Mule runtime engine (Mule) versions 4.6 and higher
+* Helm based Runtime Fabric installation or upgrade
```
```suggestion
* Helm-based Runtime Fabric installation or upgrade
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+* Mule runtime engine (Mule) versions 4.6 and higher
+* Helm based Runtime Fabric installation or upgrade
+
+Not supported features:
```
Should these be Limitations?

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+* CloudHub 2.0 Mule application deployments
+* Scheduler-based Mule applications
+* API proxy deployments
+* `rtfctl` based Runtime Fabric installation or upgrade
```
```suggestion
* `rtfctl`-based Runtime Fabric installation or upgrade
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+* `rtfctl` based Runtime Fabric installation or upgrade
+
+[IMPORTANT]
+Backup and restore functionality (during Limited GA phase) is not supported. Plan your recovery strategy accordingly, leveraging your Git repository as the source of truth for application configurations.
```
```suggestion
Backup and restore functionality (during the Limited GA phase) is not supported. Plan your recovery strategy accordingly, leveraging your Git repository as the source of truth for application configurations.
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+Before you begin, review these prerequisites and environment setup requirements.
+
+=== Installation Method
```
```suggestion
=== Installation Prerequisites
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+=== Installation Method
+
+To use Anypoint Controller, you must:
```
```suggestion
To use Anypoint Controller, you must either:
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+To use Anypoint Controller, you must:
+
+* Have an existing Runtime Fabric installation using Helm, or
```
```suggestion
* Have an existing Runtime Fabric installation using Helm.
```

hannanelson replied:
(Avoid this list structure.)

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+To use Anypoint Controller, you must:
+
+* Have an existing Runtime Fabric installation using Helm, or
+* Install or upgrade Runtime Fabric using Helm (see xref:install-helm.adoc[Install Runtime Fabric Using Helm]) or xref:upgrade-helm.adoc[Upgrade Runtime Fabric Using Helm]
```
```suggestion
* xref:install-helm.adoc[Install Runtime Fabric]) or xref:upgrade-helm.adoc[upgrade Runtime Fabric]  using Helm.
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+
+Understanding the management model is critical before deploying and managing applications with Anypoint Controller:
+
+* Applications created via Mule Application Custom Resources can only be managed through the Custom Resource itself.
```
```suggestion
* Applications created via Mule application Custom Resources can only be managed through the Custom Resource itself.
```

hannanelson commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2204 @@
+* Have an existing Runtime Fabric installation using Helm, or
+* Install or upgrade Runtime Fabric using Helm (see xref:install-helm.adoc[Install Runtime Fabric Using Helm]) or xref:upgrade-helm.adoc[Upgrade Runtime Fabric Using Helm]
+
+=== Management Constraints
```
```suggestion
=== Management Limitations
```

kevintroller committed:
Add documentation for Mule application restart feature

- Add new restart-applications.adoc page documenting the restart feature
- Include UI and API restart procedures
- Document Restart Applications permission requirement
- Note audit log tracking for restart activities
- Update navigation and managing apps index with restart feature

kevintroller committed:
Add images to restart applications documentation

- Add screenshot showing Restart Applications permission in Access Management
- Add screenshot showing Restart button in Runtime Manager
- Update formatting for API parameter sections

kevintroller committed:
Document log forwarding changes in Runtime Fabric 3.0.0

- Update use-anypoint-monitoring.adoc with sidecar-only log forwarding architecture
- Add migration steps required before upgrading to 3.0.0
- Remove am-log-forwarder role from limitations documentation
- Add notes to OpenShift installation docs about deprecated service account
- Update downgrade documentation for version 3.0.0
- Convert migration steps to proper headings with anchor links

kevintroller committed:
Changes images

kevintroller committed:
Improve clarity of scope and limitations section

- Reworded introductory sentences to better describe mixed content (deployment targets, versions, installation methods)
- Changed 'Supported features' to clarify it covers deployment targets, runtime versions, and installation methods
- Changed 'Not supported features' to explicitly state these are limitations
- Updated 'Installation Method' to 'Installation Prerequisites' for clarity
- Changed 'Management Constraints' to 'Management Limitations' for consistency
- Minor grammar and style improvements

kevintroller committed:
Update Anypoint Controller documentation and permissions

- Add target audience section for Kubernetes administrators
- Update Helm chart configuration to clarify independent settings for CRD installation and controller enablement
- Add detailed guidance for single and multiple RTF instance scenarios
- Replace Git-specific references with generic 'version control' terminology
- Update management limitations to use deployment tool terminology instead of YAML-specific language
- Add muleapplications, muleapplications/status, and httproutetemplates to rtf.mulesoft.com API group permissions
- Improve clarity and structure throughout prerequisites and configuration sections

kevintroller committed:
Clarify Anypoint Controller behavior and configuration details

- Clarify namespace synchronization label: rtf.mulesoft.com/role: workers
- Clarify controller watches namespaces with matching label rtf.mulesoft.com/agentNamespace: <rtf-namespace>
- Clarify application name exists error includes apps created via Anypoint Controller
- Rewrite Progressing and Degraded status conditions with proper YAML formatting
- Remove Pause Reconciliation section and references
- Clarify namespace flexibility and controller watching behavior
- Remove comment support statement from properties format
- Merge debug logging configuration into Check Controller Logs section
- Add cross-reference to Check Controller Logs in troubleshooting

kevintroller committed:
Update restart applications permission requirements

- Clarify that users need either Restart Applications or Create Applications permission
- Rewrite permission requirements as a simple bullet list for better readability
- Update permission granting instructions to reflect both options

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+
+After you apply this file to your Kubernetes cluster, for example by using `kubectl apply -f <filename>.yaml`, Anypoint Controller deploys the application based on the specifications.
+
+For a complete reference of all available MuleApplicationCR fields and configuration options, see xref:deploy-mule-app-anypoint-controller.adoc#understanding-custom-resource-definitions[Understanding Custom Resource Definitions].
```
```suggestion
For a complete reference of all available MuleApplication CR fields and configuration options, see xref:deploy-mule-app-anypoint-controller.adoc#understanding-custom-resource-definitions[Understanding Custom Resource Definitions].
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+  -n my-apps
+----
+[start=2]
+.Mount it in your MuleApplication CR:
```
```suggestion
. Mount it in your MuleApplication CR:
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+----
+[start=2]
+.Mount it in your MuleApplication CR:
+
```
```suggestion
+
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+Use Kubernetes native Secrets to store secret data on the Kubernetes cluster and mount them into your Mule application.
+
+. Create a Secret in your Kubernetes cluster:
+
```
```suggestion
+
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+. Copy the *Client ID* and *Client Secret*. You need these values in the next step.
+
+[IMPORTANT]
+Store the Client Secret securely. You cannot retrieve it again after leaving the create connected app page.
```
```suggestion
Store the Client Secret securely. You can't retrieve it again after leaving the create connected app page.
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+  -n <rtf-namespace>
+----
+
+Replace the following:
```
```suggestion
Replace these:
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+----
+[start=2]
+. Create an ArgoCD Application:
+
```
```suggestion
+
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+=== FluxCD Integration
+
+. Create a GitRepository resource:
+
```
```suggestion
+
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+----
+[start=2]
+. Create a Kustomization resource:
+
```
```suggestion
+
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+* The controller compares the CR spec hash with the last published hash to detect changes.
+* When changes are detected, the controller updates Anypoint Platform automatically.
+* Failed operations are retried with exponential backoff.
+* The controller periodically refreshes state even without changes (every 10 minutes).
```
```suggestion
* Synchronizes the application state every 30 seconds.
* Compares the CR spec hash with the last published hash to detect changes.
* When changes are detected, updates Anypoint Platform automatically.
* Retries failed operations with exponential backoff.
* Refreshes the state every 10 minutes, even when no changes occur.
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+
+Enable Debug Logs:
+
+By default, the log level is set to `info`, which does not emit detailed information for diagnosis. To enable debug logs for more detailed troubleshooting, add the following environment variables to the `app-config` ConfigMap and restart the agent deployment:
```
```suggestion
By default, the log level is set to `info`, which does not emit detailed information for diagnosis. To enable debug logs for more detailed troubleshooting, add these environment variables to the `app-config` ConfigMap and restart the agent deployment:
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+* <<update-a-mule-application,Update a Mule Application>>
+* <<delete-a-muleapplication,Delete a MuleApplication>>
+
+=== Update a Mule Application
```
```suggestion
[[update-a-mule-application]]
=== Update a Mule Application
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+[NOTE]
+Some fields like `target`, `businessGroup`, and `environment` are immutable and cannot be changed after creation. For a complete list of mutable and immutable fields, see xref:deploy-mule-app-anypoint-controller.adoc#understanding-custom-resource-definitions[Understanding Custom Resource Definitions].
+
+=== Delete a MuleApplication
```
```suggestion
[[delete-a-mule-application]]
=== Delete a Mule Application
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+kubectl describe pod <pod-name> -n <namespace>
+----
+
+. Common causes:
```
maybe move this to before the steps?

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+kubectl apply -f app.yaml --dry-run=server
+----
+
+. Check immutable field violations (target, businessGroup, environment cannot be changed after creation).
```
```suggestion
. Check immutable field violations (target, businessGroup, and environment can't be changed after creation).
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+Manage your application lifecycle:
+
+* <<update-a-mule-application,Update a Mule Application>>
+* <<delete-a-muleapplication,Delete a MuleApplication>>
```
```suggestion
* <<delete-a-mule-application,Delete a Mule Application>>
```

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+
+. Read the error message in the status condition.
++
+ Common errors:
```
this is rendering as a code block

sathya0 commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+   - `403 Forbidden`: Connected App lacks permissions
+   - `400 Bad Request`: Invalid configuration (for example, runtime version)
+
+. Verify configuration matches your Anypoint Platform setup.
```
```suggestion
. Verify the configuration matches your Anypoint Platform setup.
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+
+=== Delete a MuleApplication
+
+To delete an application managed by Anypoint Controller, delete the MuleApplication CR:
```
```suggestion
To delete an application managed by Anypoint Controller, delete the `muleapplication` CR:
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+The controller automatically:
+
+. Deletes the application from Anypoint Platform.
+. Removes all Kubernetes resources (Deployment, Service, ConfigMaps, Secrets).
```
```suggestion
. Removes all Kubernetes resources (Deployment, Service, ConfigMaps, and Secrets).
```

dmerlob commented on modules/ROOT/pages/deploy-mule-app-anypoint-controller.adoc:
```diff
@@ -0,0 +1,2317 @@
+. Allows Kubernetes to delete the CR.
+
+[IMPORTANT]
+Deletion is permanent. Ensure your version control repository contains the application configuration if you need to redeploy later.
```
```suggestion
Deletion is permanent. Make sure your version control repository contains the application configuration in case you need to redeploy it later.
```

kevintroller committed:
Add Anypoint Controller management clarifications

- Add note that existing applications must be redeployed as Kubernetes Custom Resources to be managed with Anypoint Controller
- Add information about Stop and Restart functionality not being supported
- Add section on configuring application URLs and ingress using Kubernetes native Ingress resources

kevintroller committed:
Update Anypoint Controller documentation with clarifications and improvements

- Add clarification that existing applications must be redeployed as Kubernetes Custom Resources
- Add note that Stop and Restart functionality is not supported
- Add section on configuring application URLs and ingress using Kubernetes native Ingress resources
- Update Runtime Manager read-only mode description to clarify buttons are grayed out
- Replace MuleApplication field reference table with cross-reference to deploy-to-runtime-fabric.adoc
- Rename 'Detailed Field Descriptions' section to 'Advanced Field Configurations'
- Fix kubectl logs commands to use correct container name (rtfd instead of anypoint-controller)

Crispy-Salesforce commented on modules/ROOT/pages/restart-applications.adoc:
```diff
@@ -0,0 +1,174 @@
+The request body must conform to the `ActionExecutionRequest` schema:
+
+* `type` (required, string): Must be set to `RESTART_APPLICATION`
+* `options` (required, object):
```
There's no description after the colon. Is it on purpose?

Crispy-Salesforce commented on modules/ROOT/pages/restart-applications.adoc:
```diff
@@ -0,0 +1,174 @@
+
+Replace `{organizationId}`, `{environmentId}`, `{access-token}`, and the values in the request body with your actual values.
+
+After a successful API call (202 Accepted response), you can monitor the application status in Runtime Manager to confirm that the restart completed successfully.
```
```suggestion
After a successful API call (202 accepted response), you can monitor the application status in Runtime Manager to confirm that the restart completed successfully.
```

Crispy-Salesforce commented on modules/ROOT/pages/use-anypoint-monitoring.adoc:
```diff
@@ -71,52 +68,104 @@ You can enable log forwarding and set log levels when you xref:deploy-to-runtime
 
-* One DaemonSet per underlying node to transport logs to Anypoint monitoring control plane.
-* One sidecar per application replica to transporting metrics to Anypoint monitoring control plane. 
+Dependent on the features you use, the sidecar agent consumes corresponding resources for optimal performance:
```
```suggestion
Depending on the features you use, the sidecar agent consumes corresponding resources for optimal performance:
```

Crispy-Salesforce commented on modules/ROOT/pages/use-anypoint-monitoring.adoc:
```diff
@@ -71,52 +68,104 @@ You can enable log forwarding and set log levels when you xref:deploy-to-runtime
+
+[NOTE]
+====
+Disabling the DaemonSet does not impact log forwarding to Anypoint Monitoring via the sidecar containers. Your applications will continue to forward logs normally after completing these steps.
```
```suggestion
Disabling the DaemonSet doesn't impact log forwarding to Anypoint Monitoring via the sidecar containers. Your applications will continue to forward logs normally after completing these steps.
```

kevintroller committed:
Clarify Kubernetes tools and deployment methods description

- Rewrite tools description to clearly state applications are deployed as Kubernetes Custom Resources
- List tools in logical order: kubectl, Helm, Terraform Kubernetes Provider, and GitOps workflows
- Remove redundant phrasing for better clarity

kevintroller committed:
Revert downgrade section to original format

- Revert changes made in commit ca21f0e9 to the downgrade section
- Remove 'Downgrading from Version 3.0.0 or Later' heading and IMPORTANT admonition
- Restore original simpler text: 'If you use AM log forwarding' instead of version-specific language
- Change step 1 back to 'Enable daemon-set log forwarding' with 'the following API'
- Change step 2 back to 'Optionally' phrasing
- Restore simple NOTE format without extra admonition wrapper
- Use 'rtf-agent' instead of 'Runtime Fabric agent' in the note

kevintroller committed:
Update terminology and improve troubleshooting section organization

- Changed 'MuleApplication CR' to 'Mule application CR' throughout
- Changed 'cannot' to 'can't' for consistency
- Changed 'the following' to 'this' or 'these' based on context
- Reorganized troubleshooting sections to show common causes/errors before resolution steps

kevintroller committed:
Update terminology and add cross-references to log forwarding documentation

- Changed 'does not' to 'doesn't' in use-anypoint-monitoring.adoc
- Added anchor ID for 'Log Forwarding in Runtime Fabric 3.0.0' section
- Added cross-references to log forwarding section in:
  * install-multiple-instances-openshift.adoc
  * install-openshift.adoc
  * limitations-self.adoc

kevintroller committed:
Update IMPORTANT note with expanded backup and restore details

- Clarified that rtfctl supports backup and restore of Mule Application CRD
- Added note that Mule Application CRs are not supported during Limited GA phase
- Emphasized using source control as single source of truth
- Added instructions to re-apply CRs after cluster restore

kevintroller committed:
Update Anypoint Controller documentation: clarify tool usage, remove incorrect note about client secret, simplify credential rotation, and update retry information description

kevintroller committed:
Remove Anypoint Controller and Restart Applications documentation

- Deleted deploy-mule-app-anypoint-controller.adoc
- Deleted restart-applications.adoc
- Removed references from nav.adoc
- Removed references from managing-apps-deployed-runtime-fabric.adoc

kevintroller committed:
Update rtfctl-commands.adoc

kevintroller committed:
Add Enable Readiness Probe on HTTP GET section

kevintroller committed:
Update customize-kubernetes-crd documentation

kevintroller committed:
Improve wording in Enable Readiness Probe section
